src/layers/fp64/extruded-choropleth-layer/extruded-choropleth-layer.js

```diff
@@ -0,0 +1,449 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Layer, assembleShaders} from '../../../lib';
+import {GL, Model, Program, Geometry} from 'luma.gl';
+import {fp64ify} from '../../../lib/utils/fp64';
+
+const glslify = require('glslify');
+
+import {vec3} from 'gl-matrix';
+
+import earcut from 'earcut';
+import flattenDeep from 'lodash.flattendeep';
+
+const ATTRIBUTES = {
+  indices: {size: 1, 0: 'index', isIndexed: true},
+  positions: {size: 4, 0: 'x', 2: 'y'},
+  heights: {size: 2, 0: 'height'},
+  normals: {size: 3, 0: 'x', 1: 'y', 2: 'z'},
+  colors: {size: 3, 0: 'red', 1: 'green', 2: 'blue'}
+};
+
+export default class ExtrudedChoroplethLayer extends Layer {
+  /**
+   * @classdesc
+   * BuildingLayer
+   *
+   * @class
+   * @param {object} props
+   * @param {bool} props.opacity=1 - default opacity is 1
+   * @param {bool} props.elevation=1 - Elevation scale
+   * @param {function} props.onHover - provides properties of the
+   *    selected building, together with the mouse event when mouse hovered
+   * @param {function} props.onClick - provide properties of the
+   *    selected building, together with the mouse event when mouse clicked
+   */
+  constructor({
+    opacity = 1,
+    elevation = 1,
+    ...props
+  }) {
+    super({
+      opacity,
+      elevation,
+      ...props
+    });
+  }
+
+  initializeState() {
+    const {gl} = this.context;
+    const {attributeManager} = this.state;
+    const {
+      elevation, color, ambientColor, pointLightColor,
+      pointLightLocation, pointLightAmbientCoefficient,
+      pointLightAttenuation, materialSpecularColor, materialShininess
+    } = this.props;
+
+    attributeManager.addDynamic(ATTRIBUTES, {
+      // Primtive attributes
+      indices: {update: this.calculateIndices},
+      positions: {update: this.calculatePositions},
+      heights: {update: this.calculateHeights},
+      colors: {update: this.calculateColors},
+      normals: {update: this.calculateNormals}
+    });
+
+    this.setState({
+      numInstances: 0,
+      model: this.getModel(gl)
+    });
+
+    this.setUniforms({
+      elevation: Number.isFinite(elevation) ? elevation : 1,
```

Comment:
This uniform never changed... so I feel it's better to put them here...ideally they should just be set once, but right now we still set them every time we draw..